data/hooks/conf_regen/06-slapd

```diff
@@ -12,56 +12,61 @@ do_init_regen() {
 
   do_pre_regen ""
 
-  # fix some permissions
-  chown root:openldap /etc/ldap/slapd.conf
-  chown -R openldap:openldap /etc/ldap/schema/
-  usermod -aG ssl-cert openldap
+  # Because slaptest can't test the LDAP config file
+  # we need to regenerate the new config and after validate it
+  # regenerate LDAP config directory from slapd.ldif
+  rm -Rf /etc/ldap/slapd_new.d
+  mkdir /etc/ldap/slapd_new.d
+  slapadd -n0 -l /etc/ldap/slapd.ldif -F /etc/ldap/slapd_new.d/ 2>&1
 
   # check the slapd config file at first
-  slaptest -Q -u -f /etc/ldap/slapd.conf
+  slaptest -Q -u -F /etc/ldap/slapd_new.d
 
-  # regenerate LDAP config directory from slapd.conf
+  # Move to the new config
   rm -Rf /etc/ldap/slapd.d
-  mkdir /etc/ldap/slapd.d
-  slaptest -f /etc/ldap/slapd.conf -F /etc/ldap/slapd.d/ 2>&1
+  mv /etc/ldap/slapd_new.d /etc/ldap/slapd.d
+
+  # fix some permissions
+  chown root:openldap /etc/ldap/slapd.ldif
+  chown -R openldap:openldap /etc/ldap/schema/
   chown -R openldap:openldap /etc/ldap/slapd.d/
+  usermod -aG ssl-cert openldap
 
   service slapd restart
 }
 
 do_pre_regen() {
   pending_dir=$1
 
-  cd /usr/share/yunohost/templates/slapd
-
-  # create needed directories
-  ldap_dir="${pending_dir}/etc/ldap"
-  schema_dir="${ldap_dir}/schema"
-  mkdir -p "$ldap_dir" "$schema_dir"
-
-  # remove legacy configuration file
-  [ ! -f /etc/ldap/slapd-yuno.conf ] \
-    || touch "${pending_dir}/etc/ldap/slapd-yuno.conf"
-
   # remove temporary backup file
   rm -f "$tmp_backup_dir_file"
 
-  # retrieve current and new backends
+  # Define if we need to migrate from hdb to mdb
   curr_backend=$(grep '^database' /etc/ldap/slapd.conf 2>/dev/null | awk '{print $2}')
-  new_backend=$(grep '^database' slapd.conf | awk '{print $2}')
-
-  # save current database before any conf changes
-  if [[ -n "$curr_backend" && "$curr_backend" != "$new_backend" ]]; then
+  if [ -e /etc/ldap/slapd.conf ] && [ -n "$curr_backend" ] && \
```

Comment:
```suggestion
  if [ -e /etc/ldap/slapd.ldif ] && [ -n "$curr_backend" ] && \
```

Reply:
Eeeh I think that's precisely the legacy file used to check the database format ... Though in fact I don't even know if it's worth grepping '^database' since the file doesn't exist after the automigration ... But maybe Josue knows more about this

Reply:
Well, as I understood in the new config file format we don't need to specify the type of the database.

Here it's just a check of a really old legacy thing that I was really attempted to drop. It's the migration of the database format from hdb to mdb. So yes we need to keep this as it is.